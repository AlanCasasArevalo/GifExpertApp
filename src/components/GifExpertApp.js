import React, {useState} from 'react'
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories)

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory
                setCategories={setCategories}
            />
            <hr/>

            <ol>
                {
                    categories.map(category =>
                        (<GifGrid
                                key={category}
                                category={category}
                            />
                        ))
                }
            </ol>

        </>
    )
}
//
// GifExpertApp.propTypes = {
//     value: PropTypes.number
// }

export default GifExpertApp;

