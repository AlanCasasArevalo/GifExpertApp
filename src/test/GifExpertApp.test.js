import {shallow} from "enzyme";
import React from "react";
import GifExpertApp from "../components/GifExpertApp";

describe('GifExpertApp Component test', () => {

    let wrapper = shallow(
        <GifExpertApp
        />)

    beforeEach(() => {
        wrapper = shallow(
            <GifExpertApp
            />)
    })

    test('GifExpertApp should be defined', () => {
        expect(<GifExpertApp/>).toBeDefined()
    })
    test('GifExpertApp should be an object', () => {
        expect(typeof <GifExpertApp/>).toBe('object')
    })

    test('GifExpertApp should return a snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })


    test('GifExpertApp should return a category list', () => {
        const categories = ['Hello', 'Bye']
        wrapper = shallow(
            <GifExpertApp
                defaultCategories={categories}
            />)
        expect(wrapper).toMatchSnapshot()
        const length = wrapper.find('GifGrid').length
        expect(length).toBe(categories.length)
    })

    test('GifExpertApp should get h1', () => {
        const title = wrapper.find('h1').props().children

        expect('GifExpertApp').toStrictEqual(title)
    })

    test('GifExpertApp should get ol', () => {
        const ol = wrapper.find('ol').props().children

        expect(ol).toBeTruthy()
    })
})
