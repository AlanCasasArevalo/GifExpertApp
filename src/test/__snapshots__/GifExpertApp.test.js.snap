// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Component test GifExpertApp should return a category list 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Hello"
      key="Hello"
    />
    <GifGrid
      category="Bye"
      key="Bye"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp Component test GifExpertApp should return a snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
